src/emulator/dataconnectEmulator.ts

```diff
@@ -90,27 +89,28 @@ export class DataConnectEmulator implements EmulatorInstance {
     });
     this.usingExistingEmulator = false;
     if (this.args.autoconnectToPostgres) {
+      const pgPort = this.args.postgresListen?.[0].port;
+      const pgHost = this.args.postgresListen?.[0].address;
       let connStr = dataConnectLocalConnString();
       if (dataConnectLocalConnString()) {
         this.logger.logLabeled(
           "INFO",
           "Data Connect",
           `FIREBASE_DATACONNECT_POSTGRESQL_STRING is set to ${dataConnectLocalConnString()} - using that instead of starting a new database`,
         );
-      } else {
+      } else if (pgPort && pgPort) {
         const pgServer = new PostgresServer(dbId, "postgres");
-        if (this.args.postgresPort) {
-          const process = await lsofi(this.args.postgresPort);
+        if (this.args.postgresListen) {
+          const process = await lsofi(pgPort);
           if (process) {
             const errMessage =
-              `Data Connect: Unable to start PGLite server on port ${this.args.postgresPort} because it is already in use by process number ${process}. This may occur if you are running another instance of Postgres.` +
+              `Data Connect: Unable to start PGLite server on port ${pgPort} because it is already in use by process number ${process}. This may occur if you are running another instance of Postgres.` +
               ` You can choose a different port by setting 'firebase.json#emulators.dataconnect.postgresPort', or you can unset that field to automatically find an open port.`;
             throw new FirebaseError(errMessage);
           }
         }
-        const port = this.args.postgresPort || (await findOpenPort(5432));
-        const server = await pgServer.createPGServer(this.args.postgresHost, port);
-        connStr = `postgres://${this.args.postgresHost ?? "127.0.0.1"}:${port}/${dbId}?sslmode=disable`;
+        const server = await pgServer.createPGServer(pgHost, pgPort);
+        connStr = `postgres://${pgHost}:${pgPort}/${dbId}?sslmode=disable`;
```

Comment:
Please use `connectableHostname` to handle `pgHost` being 0.0.0.0 etc and quote IPv6 addresses. Similar to https://github.com/firebase/firebase-tools/blob/0e2ab54c522d40ec9fe874e108332427e4886e0a/src/emulator/registry.ts#L187-L192

You may want to use the `new URL`  constructor FWIW

Reply:
Done